### PR TITLE
Support Cairo 0.10.0

### PR DESCRIPTION
**Pull request by juniset**
This PR:
- adds support for 0.10.0
- adopts the OZ casing convention where external methods are camel case
- adds some basic test
- renames `Signer.cairo` into `StarkSigner.cairo` to prepare for more signer plugins

### CODE REVIEW

File: tests/test_stark_signer.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return starknet.state, account, sts_plugin_decl.class_hash
+
+@pytest.fixture
+def contract_factory(contract_classes, contract_init):
+    account_cls, sts_plugin_cls = contract_classes
+    state, account, sts_plugin_hash = contract_init
+    _state = state.copy()
+
+    account = cached_contract(_state, account_cls, account)
+    account_as_plugin = cached_contract(_state, sts_plugin_cls, account)
```
Comment (CremaFR) [inline tests/test_stark_signer.py:42]: Smart, never though we could do this to use plugin's getters on the account state

File: contracts/plugins/signer/StarkSigner.cairo [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        signature_r=sig_r,
+        signature_s=sig_s);
+
+    return (is_valid=TRUE);
+}
+
+func assert_only_self{syscall_ptr: felt*}() -> () {
+    let (self) = get_contract_address();
+    let (caller_address) = get_caller_address();
+    with_attr error_message("account: only self") {
```
Comment (CremaFR) [inline contracts/plugins/signer/StarkSigner.cairo:111]:
```suggestion
    with_attr error_message("starksigner: only self") {
```

Shouldn't this also be a starksigner message?

File: contracts/plugins/signer/StarkSigner.cairo [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+from contracts.account.library import CallArray
+
+@storage_var
+func StarkSigner_public_key() -> (res: felt) {
+}
+
+@external
+func initialize{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(plugin_data_len: felt, plugin_data: felt*) {
+    let (is_initialized) = StarkSigner_public_key.read();
+    with_attr error_message("starksigner: already initialized") {
```
Comment (CremaFR) [inline contracts/plugins/signer/StarkSigner.cairo:17]:
```suggestion
    with_attr error_message("StarkSigner: already initialized") {
```

Should we use the same case the contract name is using?

**Review by CremaFR — APPROVED** [review]
LGTM, I just have a few comment about error message format

File: contracts/plugins/signer/StarkSigner.cairo (hunk 3 above)
Comment (juniset) [reply contracts/plugins/signer/StarkSigner.cairo]: I've updated all error messages to use the contract name casing.

**Pull request merged**